File: src/com/company/transportoPriemones/Dviratis.java
package com.company.transportoPriemones;

import com.company.kainosSkaiciavimas.Ikainiai;
import com.company.kainosSkaiciavimas.IkainiaiBuilder;
import com.company.strategijos.TransportoPasirinkimoStrategija;

public class Dviratis implements TransportoPasirinkimoStrategija {
    private int priemoneId;

    @Override
    public String pasirinktiTransportoPriemone(int priemoneId) {
        //if(priemoneID nepasirinktas) pasirinkti
        this.priemoneId = priemoneId;
        return "Rambynas 300";
        //else return null;
    }

    @Override
    public Ikainiai gautiTransportoPriemonesIkainius() {
        return new IkainiaiBuilder()
                .setPradineKaina(10)
                .setKilometroKaina(0)
                .setMinutesKaina(0)
                .setVirsytoLaikoKaina(20)
                .setVirsytoAtstumoKaina(0)
                .setParosKaina(300)
                .setSavaitesKaina(1500)
                .setMenesioKaina(3000)
                .createIkainiai();
    }

    @Override
    public void paliktiTransportoPriemone() {
        //palikti priemoneId
        System.out.println("Paliktas Rambynas 300, id: " + priemoneId);
    }

    @Override
    public void pranestiApieNetiketuma() {
        System.out.println("Dekojame uz pranesta dviracio Rambynas 300, id: " + priemoneId + " apgadinima.");
    }
}

File: Mixins/src/com/company/keliones/Dviratis.scala
package com.company.keliones

import com.company.kainosSkaiciavimas.Ikainiai

trait Dviratis extends TransportoPriemone {
  private val transportoPriemonesPavadinimas = "Rambynas 300"
  val _ikainiai: Ikainiai = new Ikainiai(10, 0, 0, 20, 0, 300, 1500, 3000)

  def gautiTransportoPriemonesPavadinima: String = transportoPriemonesPavadinimas

  def gautiIkainius: Ikainiai = _ikainiai

  def pranestiApieNetiketuma(): Unit = System.out.println("Dekojame uz pranesta dviracio Rambynas 300 apgadinima.")

  protected def paliktiTransportoPriemone(priemoneId : Int): Unit = { //palikti priemoneId
    System.out.println("Paliktas Rambynas 300, id: " + priemoneId)
  }
}